pkg/planner/core/memtable_infoschema_extractor.go

```diff
@@ -721,6 +739,13 @@ func findSchemasForTables(
 			remains = append(remains, tbl)
 		}
 	}
+	sort.Slice(schemaSlice, func(i, j int) bool {
```

Comment:
Can we create a new slice to store tables instead of directly modifing input slice?

Reply:
The input slice is not used by anyone else. Let me add some comments for this.

Reply:
https://github.com/pingcap/tidb/pull/56378/commits/291d8cc58a967e9e2f13c710f2573ec712fe4b1c